addons/hr_recruitment/models/hr_recruitment.py

```diff
@@ -360,6 +360,17 @@ def message_new(self, msg, custom_values=None):
             through message_process.
             This override updates the document according to the email.
         """
+
+        # Check if already exists an application for the email address
+        # and post a message to every application found matching the email
+        existing = self.search([
+            '|',
+            ('email_from', 'ilike', '<' + msg.get('from') + '>'),
```

Comment:
If the key `from` is not in the dict,  `'<' + msg.get('from') + '>'` would raise an exception because strings cannot be concatenated with `None`. Please use string interpolation: ` '<%s>' % msg.get('from')`